test/snapshot.test.js

```diff
@@ -0,0 +1,93 @@
+const fs = require('fs');
+const test = require('tap').test;
+const sinon = require('sinon');
+const path = require('path');
+
+const config = require('../lib/config');
+const snapshotReader = require('../lib/snapshot/reader');
+
+config.initConfig({projectId: 'whatever'});
+
+test('snapshot reader defaults to repo snapshot when bundled is missing', async (t) => {
+  const repoSnapshotStub = [{
+    "methodId": {
+      "className": null,
+      "filePath": "mime.js",
+      "methodName": "Mime.prototype.lookup"
+    },
+    "packageName": "mime",
+    "version": ["<1.4.1"]
+  }];
+  const stub = sinon.stub(fs, 'readFileSync');
+  stub.withArgs(config.functionPaths.repo.snapshot).returns(JSON.stringify(repoSnapshotStub));
+
+  snapshotReader.loadFromLocal();
+  const result = snapshotReader.getLatest();
+
+  t.deepEqual(Object.keys(result), ['mime']);
+  t.deepEqual(Object.keys(result['mime']), ['mime.js']);
+  stub.restore();
+  t.end();
+});
+
+function readerFallsBackToRepoSnapshotWhenBundledError(t, bundledResponse) {
+  const repoSnapshotStub = [{
+    "methodId": {
+      "className": null,
+      "filePath": "mime.js",
+      "methodName": "Mime.prototype.lookup"
+    },
+    "packageName": "mime",
+    "version": ["<1.4.1"]
+  }];
+
+  config.functionPaths.bundle.snapshot = path.join(__dirname, './fixtures/snapshots/bundled-snapshot.json'); //Just needs to point to an existing file
+  const stub = sinon.stub(fs, 'readFileSync');
+  stub.withArgs(config.functionPaths.repo.snapshot).returns(JSON.stringify(repoSnapshotStub));
+  stub.withArgs(config.functionPaths.bundle.snapshot).returns(bundledResponse);
+
+  const existsStub = sinon.stub(fs, 'existsSync').returns(true);
+
+  snapshotReader.loadFromLocal();
+  const result = snapshotReader.getLatest();
+
+  t.deepEqual(Object.keys(result), ['mime']);
+  t.deepEqual(Object.keys(result['mime']), ['mime.js']);
+  
+  stub.restore();
+  existsStub.restore();
+}
+test('snapshot reader falls back to repo snapshot when bundled errors', async (t) => {
```

Comment:
I would add another case here:
a snapshot broken in the middle somehow, so it does contain some entries which would look OK to the eye, but something else would break the parsing.
see what could break `processFunctionsToInspect`, preferably a valid JSON, and use it.
then, in `readerFallsBackToRepoSnapshotWhenBundledError` validate that the contents of that JSON aren't instrumented, or something like that